cri-o crashing on restart

**Description**

Restarting cri-o after creating a single sandbox causes cri-o to crash with a panic.

**Steps to reproduce the issue:**

1. Start cri-o
2. Create a sandbox using `crictl runp` and this config:
```
 {
      "metadata": {
          "name": "nginx-sandbox",
          "namespace": "default",
          "attempt": 1,
          "uid": "hdishd83djaidwnduwk28bcsb"
      },
      "log_directory": "/tmp",
      "linux": {
      }
  }
```
3. Restart cri-o

**Describe the results you received:**

cri-o crashes with this in the logs:

```
NFO[2020-04-02 09:33:53.984489191Z] Update default CNI network name to crio       file="ocicni/ocicni.go:375"
DEBU[2020-04-02 09:33:53.986316735Z] Golang's threads limit set to 57060           file="server/server.go:291"
WARN[2020-04-02 09:33:53.988149010Z] Could not restore sandbox 25dd18310936e5099c98d92affc41472d8a55b3a6e248382e89836d35852b38d container 25dd18310936e5099c98d92affc41472d8a55b3a6e248382e89836d35852b38d: unexpected end of JSON input  file="server/server.go:188"
WARN[2020-04-02 09:33:53.989093296Z] Unable to delete container k8s_POD_nginx-sandbox_default_hdishd83djaidwnduwk28bcsb_1: unlinkat /var/run/containers/storage/overlay-containers/25dd18310936e5099c98d92affc41472d8a55b3a6e248382e89836d35852b38d/userdata/shm: device or resource busy  file="server/server.go:191"
WARN[2020-04-02 09:33:53.989917887Z] Unable to delete container k8s_nginx-sandbox_default_hdishd83djaidwnduwk28bcsb_1: identifier is not a container  file="server/server.go:191"
WARN[2020-04-02 09:33:53.989964625Z] Deleting all containers under sandbox 25dd18310936e5099c98d92affc41472d8a55b3a6e248382e89836d35852b38d since it could not be restored  file="server/server.go:201"
DEBU[2020-04-02 09:33:53.990103525Z] Sandboxes: []                                 file="server/server.go:418"
DEBU[2020-04-02 09:33:53.991350205Z] Registered SIGHUP watcher for config          file="config/reload.go:32"
DEBU[2020-04-02 09:33:53.991553502Z] Monitoring "/usr/share/containers/oci/hooks.d" for hooks  file="hooks/monitor.go:43"
panic: attempted to update last-writer in lockfile without the write lock

goroutine 64 [running]:
github.com/containers/storage/pkg/lockfile.(*lockfile).Touch(0xc000349540, 0xc0005fc960, 0x0)
        /root/cri-o/vendor/github.com/containers/storage/pkg/lockfile/lockfile_unix.go:193 +0x235
github.com/containers/storage.(*containerStore).Touch(0xc00049e540, 0x3e, 0xc000444050)
        /root/cri-o/vendor/github.com/containers/storage/containers.go:588 +0x33
github.com/containers/storage.(*containerStore).Save(0xc00049e540, 0x0, 0x0)
        /root/cri-o/vendor/github.com/containers/storage/containers.go:228 +0x202
github.com/containers/storage.(*containerStore).Delete(0xc00049e540, 0xc00005ae00, 0x39, 0x0, 0x0)
        /root/cri-o/vendor/github.com/containers/storage/containers.go:405 +0x26c
github.com/containers/storage.(*store).DeleteContainer.func2(0xc0001013e0, 0x21c56c0, 0xc00049e540, 0xc00005ae00, 0x39, 0xc0005ef640)
        /root/cri-o/vendor/github.com/containers/storage/store.go:2335 +0x49
created by github.com/containers/storage.(*store).DeleteContainer
        /root/cri-o/vendor/github.com/containers/storage/store.go:2334 +0x540
```

**Describe the results you expected:**

cri-o successfully restarts and restores the pod sandbox

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `crio --version`:**

```
crio version 
Version:       1.18.0-dev
GitCommit:     7e8ccf9364af5138daf83fc22a1e1d47e42a8e90
GitTreeState:  dirty
BuildDate:     2020-04-01T13:22:18Z
GoVersion:     go1.14.1
Compiler:      gc
Platform:      linux/amd64
Linkmode:      dynamic
```

but also happens with 1.16 and 1.17

cc @saschagrunert

--- a/internal/lib/container_server.go
+++ b/internal/lib/container_server.go
@@ -151,11 +151,11 @@ func (c *ContainerServer) LoadSandbox(id string) error {
 	}
 	var m rspec.Spec
 	if err := json.Unmarshal(config, &m); err != nil {
-		return err
+		return errors.Wrap(err, "error unmarshalling sandbox spec")
 	}
 	labels := make(map[string]string)
 	if err := json.Unmarshal([]byte(m.Annotations[annotations.Labels]), &labels); err != nil {
-		return err
+		return errors.Wrapf(err, "error unmarshalling %s annotation", annotations.Labels)
 	}
 	name := m.Annotations[annotations.Name]
 	name, err = c.ReservePodName(id, name)
@@ -169,7 +169,7 @@ func (c *ContainerServer) LoadSandbox(id string) error {
 	}()
 	var metadata pb.PodSandboxMetadata
 	if err := json.Unmarshal([]byte(m.Annotations[annotations.Metadata]), &metadata); err != nil {
-		return err
+		return errors.Wrapf(err, "error unmarshalling %s annotation", annotations.Metadata)
 	}
 
 	processLabel := m.Process.SelinuxLabel
@@ -179,19 +179,19 @@ func (c *ContainerServer) LoadSandbox(id string) error {
 
 	kubeAnnotations := make(map[string]string)
 	if err := json.Unmarshal([]byte(m.Annotations[annotations.Annotations]), &kubeAnnotations); err != nil {
-		return err
+		return errors.Wrapf(err, "error unmarshalling %s annotation", annotations.Annotations)
 	}
 
 	portMappings := []*hostport.PortMapping{}
 	if err := json.Unmarshal([]byte(m.Annotations[annotations.PortMappings]), &portMappings); err != nil {
-		return err
+		return errors.Wrapf(err, "error unmarshalling %s annotation", annotations.PortMappings)
 	}
 
 	privileged := isTrue(m.Annotations[annotations.PrivilegedRuntime])
 	hostNetwork := isTrue(m.Annotations[annotations.HostNetwork])
 	nsOpts := pb.NamespaceOption{}
 	if err := json.Unmarshal([]byte(m.Annotations[annotations.NamespaceOptions]), &nsOpts); err != nil {
-		return err
+		return errors.Wrapf(err, "error unmarshalling %s annotation", annotations.NamespaceOptions)
 	}
 
 	sb, err := sandbox.New(id, m.Annotations[annotations.Namespace], name, m.Annotations[annotations.KubeName], filepath.Dir(m.Annotations[annotations.LogPath]), labels, kubeAnnotations, processLabel, mountLabel, &metadata, m.Annotations[annotations.ShmPath], m.Annotations[annotations.CgroupParent], privileged, m.Annotations[annotations.RuntimeHandler], m.Annotations[annotations.ResolvPath], m.Annotations[annotations.HostName], portMappings, hostNetwork)

--- a/server/sandbox_run_linux.go
+++ b/server/sandbox_run_linux.go
@@ -185,13 +185,12 @@ func (s *Server) runPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 	}
 
 	// Add special container name label for the infra container
-	labelsJSON := []byte{}
 	if labels != nil {
 		labels[types.KubernetesContainerNameLabel] = leaky.PodInfraContainerName
-		labelsJSON, err = json.Marshal(labels)
-		if err != nil {
-			return nil, err
-		}
+	}
+	labelsJSON, err := json.Marshal(labels)
+	if err != nil {
+		return nil, err
 	}
 
 	// add annotations

